Add MF.Config.MAX_VARS_PER_FRAME to limit number of variables read in each frame

DocMoebiuz opened the pull request:
This is a new PR re-using the PR from here https://github.com/MobiFlight/MobiFlight-WASM-Module/pull/11

* uses std::vector instead of std::list
* `MOBIFLIGHT_MAX_VARS_PER_FRAME` has a default value of 30, but it can be set to a new value using the command `MF.Config.MAX_VARS_PER_FRAME.Set.`

Koseng commented:
Hi Sebastian,

I also did an implementation, than still wanted to add a new command and more stuff and lost track of it.

Attached my last version, which I also tested. I have not yet looked at your code, will be interested to see how similar it will be ;-). My code also uses std::vector and defaulted to 30.

Attached my code:
[Module.zip](https://github.com/MobiFlight/MobiFlight-WASM-Module/files/9283972/Module.zip)

Koseng commented:
Ok, I checked the code now and compared the solutions:
It is pretty similar. Sorry that I lost track, otherwise one of us could have saved some time...

Remarks:
- I added the setting for the maximum variables per frame not globally, but as a setting of each client. So in a multi client scenario each client can decide for itself. For example 1 client does not care it is just using 5 variables. A second client has a lot of variables and wants to set its limit to xy for some reason.
- And a second small optimization I added was the AbortReading. So when ClearSimVars is executed, it will quickly stop to read further variables as soon as possible. Since we can't use locks it is not perfect but a small optimization to make a conflict less probable.

DocMoebiuz commented:
> Ok, I checked the code now and compared the solutions: It is pretty similar. Sorry that I lost track, otherwise one of us could have saved some time...
> 
> Remarks:
> 
> * I added the setting for the maximum variables per frame not globally, but as a setting of each client. So in a multi client scenario each client can decide for itself. For example one client does not care it is just using 5 variables. A second client has a lot of variables and wants to set its limit to xy for some reason.
> * And a second small optimization I added was the AbortReading. So when ClearSimVars is executed, it will quickly stop to read further variables as soon as possible. Since we can't use locks it is not perfect but a small optimization to make a conflict less probable.

i don't think a client specific limit makes sense because your tests show that the total number of variables read per frame is the one that matters. I don't think that we need more or less than 30 per frame - maybe 40 - but i don't expect that anyone will adjust it  - i am not planning to make it configurable for Mobiflight either. I will wait for the community feedback first - and i might simply adjust the default value with one of the next releases. 

what is your 2nd idea - I am not sure if i understand right.

Koseng commented:
Just check the attached code in the first comment and look for AbortReading.

Koseng commented:
The AbortReading was under the assumption that the Dispatch Callback and the external Simconnect Client calls might run concurrently. I am not sure whether this is the case or completely everything is queued and runs in one thread. If there is no concurrency, then the AbortReading stuff makes no sense.

DocMoebiuz commented:
@Koseng did you every experiment with the precompile method for RPN? Did that make a difference?

Koseng commented:
Yes, I tried and in my experiments it made no difference.

DocMoebiuz commented:
what's you discord username?

Koseng commented:
Wrote you in discord.
